apps/backoffice-v2/src/common/hooks/useHomeLogic/useHomeLogic.ts

```diff
@@ -0,0 +1,21 @@
+import { useSearchParams } from 'react-router-dom';
+import { useZodSearchParams } from '@/common/hooks/useZodSearchParams/useZodSearchParams';
+import { HomeSearchSchema } from '@/pages/Home/home-search-schema';
+
+export const useHomeLogic = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [dateRange, setDateRange] = useZodSearchParams(HomeSearchSchema);
```

Comment:
Remember we pass `HomeSearchSchema` since it describes not only the date range.
```suggestion
  const [searchParams, setSearchParams] = useZodSearchParams(HomeSearchSchema);
```